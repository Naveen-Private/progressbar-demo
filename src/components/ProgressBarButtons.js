import React from 'react'

const ProgressBarButtons = (props) => {
    return(
        props.progressBarButtons.map((progressBar, index) =>
            <option key={index} value={index}>ProgressBar {index+1}</option>
        )
    )
}

export default ProgressBarButtons