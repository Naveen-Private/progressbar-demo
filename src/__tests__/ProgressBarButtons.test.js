import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import ProgressBarButtons from '../components/ProgressBarButtons'

describe('ProgressBarButtons', () => {
    it('should be defined', () => {
      expect(ProgressBarButtons).toBeDefined();
    });
    it('should render correctly', () => {
      const tree = shallow(
        <ProgressBarButtons progressBarButtons={[10, 20, -30]} />
      );
      expect(tree).toMatchSnapshot();
    });
});