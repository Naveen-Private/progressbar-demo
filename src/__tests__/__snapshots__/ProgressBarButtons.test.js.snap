// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBarButtons should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProgressBarButtons
    progressBarButtons={
        Array [
            10,
            20,
            -30,
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Array [
    <option
      value={0}
>
      ProgressBar 
      1
</option>,
    <option
      value={1}
>
      ProgressBar 
      2
</option>,
    <option
      value={2}
>
      ProgressBar 
      3
</option>,
  ],
  Symbol(enzyme.__nodes__): Array [
    Array [
      <option
        value={0}
>
        ProgressBar 
        1
</option>,
      <option
        value={1}
>
        ProgressBar 
        2
</option>,
      <option
        value={2}
>
        ProgressBar 
        3
</option>,
    ],
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
